### ISSUE TITLE
New Feature: DateTime/Number format suggestion

### ISSUE THREAD

**Issue by ravero** [mfractor/mfractor-feedback#280]
.NET has a rich formatting system for strings that is almost a DSL itself. I've always find myself checking on the [Standard numeric format strings](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings) and [Standard date/time format strings](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings) docs for quick reference.

Going back and forward this docs is tedious and time consuming. It would be best if we could add Intellisense extensions to MFractor to provide suggestions when in a format string. The image below is taken from Rider and is a pretty good example of this implementation.

![screen_shot_2020-07-12_at_20 16 56](https://user-images.githubusercontent.com/1834963/88211722-7ca46f80-cc2c-11ea-820c-65b0e80bdd43.png)

Notice that it appears as you time the argument to ToString on a DateTime type. This is the first place where it should appear. Another very useful place to have this Intellisense would be in the `StringFormat` property of a Binding Expression.

Additional related suggestions:
- Implement a code analysis to check if format strings are in correct format
- Implement an adornment to appear on the right side of a format string with an example of how a date or a number would be formatted with that string, so the developer can immediately check the result of its format expression

**Comment by matthewrdev** [comment]
@ravero I believe following our discussion we concluded this was included in Visual Studio by default now.

Can you confirm and then close this ticket if so?

**Comment by ravero** [comment]
@matthewrdev this is true for C# `ToString()`, but its still lacking on XAML Binding `StringFormat` parameter, where I think it would be really useful. Can we keep it under consideration for this scenario?
